### PR TITLE
#88 feature: A buyer should be able to delete product and clear cart

### PR DESCRIPTION
**Pull request by oliviertech212**
## What does this PR do?
- buyer should be able to clear cart and update total
## Description of Task to be completed
- implement Reduxtoolkit
- clear buyer's cart
- remove single item from cart
- update total

## How should this be manually tested?
- clone the repo `git clone https://github.com/atlp-rwanda/ecommerce-app-champions-fe.git`
- checkout to `ft-clear-cart-cp-88`
- create `.env `file and paste variables from `.env.example`
- install the packages by `npm install`
- start the server  `npm run dev`
- login in swagger documentation as a buyer. to add product to cart
-  login as `buyer`
- navigate to `cart` 
Screenshot 2023-05-11 at 08 48 15
-then you can remove single product or clear cart. to remove all product

image

## Trello card 88

### CODE REVIEW

**Comment by oliviertech212** [comment]
[A buyer should be able to clear their cart](https://trello.com/c/7UmUtmtE/88-a-buyer-should-be-able-to-clear-their-cart)

**Review by wakaflorien — CHANGES_REQUESTED** [review]
- when the cart is empty shipping still show $5

image

File: src/components/cart/CartCheckout.jsx
Comment (sostene12) [inline src/components/cart/CartCheckout.jsx:1]: remove coments

**Review by sostene12 — CHANGES_REQUESTED** [review]
- when i click on remove product from the cart, it does not work instead it loads clear cart button

**Review by sostene12 — CHANGES_REQUESTED** [review]
- when i click on remove product from the cart, it does not work instead it loads clear cart button

**Review by sostene12 — CHANGES_REQUESTED** [review]
- when i click on remove product from the cart, it does not work instead it loads clear cart button

**Review by sostene12 — CHANGES_REQUESTED** [review]
- when i click on remove product from the cart, it does not work instead it loads clear cart button

**Review by patsicko — APPROVED** [review]

**Review by wakaflorien — CHANGES_REQUESTED** [review]
- Hide the checkout/side card when the cart is empty

**Review by wakaflorien — APPROVED** [review]

**Review by sostene12 — COMMENTED** [review]
@oliviertech212 you can not display the loader when there is no token, so the user is not logged in you can redirect to login or display other thinks.

**Review by sostene12 — APPROVED** [review]

**Pull request merged**